aipg/sandbox/service.py

```
@@ -0,0 +1,29 @@
+
+    This service treats the runner as a port and performs minimal validation,
+    delegating execution details to the adapter.
+    """
+
+    def __init__(self, runner: SandboxRunner, default_timeout_seconds: int = 5) -> None:
+        self._runner = runner
+        self._default_timeout_seconds = default_timeout_seconds
+
+    def run_code(self, code: str, input_data: Optional[str] = None, timeout_seconds: Optional[int] = None) -> SandboxResult:
+        if not isinstance(code, str) or code.strip() == "":
+            raise ValueError("code must be a non-empty string")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)